packages/ui-keyring/src/Keyring.ts

```diff
@@ -195,12 +195,17 @@ export class Keyring extends Base implements KeyringStruct {
       return;
     }
 
-    const address = this.encodeAddress(
-      isHex(json.address)
-        ? hexToU8a(json.address)
-        // FIXME Just for the transition period (ignoreChecksum)
-        : this.decodeAddress(json.address, true)
-    );
+    // We assume anything hex that is not 32bytes (64 + 2 bytes hex) is an Ethereum-like address
+    // (this caters for both H160 addresses as well as full or compressed publicKeys) - in the case
+    // of both ecdsa and ethereum, we keep it as-is
+    const address = isHex(json.address) && json.address.length !== 66
+      ? json.address
+      : this.encodeAddress(
```

Comment:
This does look werid, since we don't recode. But the recode is actually a migration step, which is not needed here. (Right at the start the address was actually the hex publicKey, which is why this was added. It is still quite "flakey" since the keys are actually hex, but the address an actual on-chain address)